backend/functions/src/emailing/functions.ts

```diff
@@ -0,0 +1,89 @@
+import axios from 'axios'
+
+const GRAPH_ENDPOINT = 'https://graph.microsoft.com'
+
+/* Add Recipients parses frontend data and adds to 
+    the 'toRecipients' key-value pair of 
+    the email request body sent to the MS Graph API */
+function addRecipients(messageBody: any, rcpts = []) {
+  const cloned = Object.assign({}, messageBody)
+  if (rcpts.length > 0) {
+    cloned.message['toRecipients'] = createRecipients(rcpts)
+  }
+  return cloned
+}
+
+/* Create Recipients parses the rcpts (string list of student emails) 
+    sent from the frontend. Called in addRecipients. */
+function createRecipients(rcpts: string[]) {
+  return rcpts.map((rcpt) => {
+    return {
+      emailAddress: {
+        address: rcpt,
+      },
+    }
+  })
```

Comment:
Personally I like omitting the return statement and mapping to a new value directly in cases like these:
```js
.map((rcpt) => ({ emailAddress: { address: rcpt } })
```
Your way is fine tho